docs/source/deployment/lambda.md

```diff
@@ -58,6 +59,226 @@ exports.graphqlHandler = server.createHandler();
 
+## Deploying with the AWS Serverless Application Model (SAM)
+
+The [AWS Serverless Application Model (AWS SAM)](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html) is an open-source framework that you can use to build serverless applications on AWS.
+
+### Configuring your SAM template.yaml
+
+An AWS SAM template file closely follows the format of an AWS CloudFormation template file, which is described in [Template anatomy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-anatomy.html) in the AWS CloudFormation User Guide.
+
+For the sake of this example, the following file can just be copied and pasted into the root of your project.
+
+```
+# template.yaml
+
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+
+Resources:
+  GraphQLFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: graphql/
+      Handler: app.graphqlHandler
+      Timeout: 5 #
+      MemorySize: 256
+      Runtime: nodejs14.x
+      Events:
+        AutomaticPersistedQueries:
+          Type: Api
+          Properties:
+            Path: /
+            Method: get
+        GraphQL:
+          Type: Api
+          Properties:
+            Path: /
+            Method: post
+        ApolloSandbox:
+          Type: Api
+          Properties:
+            Path: /
+            Method: options
+```
+
+You will need to ensure `options` are exposed to use Apollo Explorer and Apollo Sandbox.
+
+*Note: The example above is using v1 of AWS API Gateway which is denoted by the `Type: Api`, but v2 (`Type: HttpApi`) is supported as well. The `apollo-server-lambda` [repository currently has tests that cover both v1 and v2 API Gateway events and ALB events](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server-lambda/src/__tests__/lambdaApollo.test.ts). More info about API Event Source can be found [here](https://github.com/aws/serverless-application-model/blob/master/versions/2016-10-31.md#api)*
+
+### Invoking SAM locally
+
+To run your project locally with SAM, you will need to provide an event payload that "invokes" the AWS Lambda handler running locally in docker on your machine. [Invoking your lambda function](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-invoke.html) locally can be done with the SAM CLI:
+
+```
+sam local invoke "GraphQLFunction" -e event.json
+```
+
+For the `event.json`, you will need to provide the proper event depending on what is being used to provide the event to the AWS lambda. This will most commonly used will be an [AWS API Gateway, either v1 or v2](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html).
+
+Below is an example of the minimum `event.json` needed for a standard GraphQL request when using AWS API Gateway v1:
```

Comment:
Why is it helpful to document both formats?

Reply:
It's just that they're very different. Invoking a lambda has to happen with a json payload and the GraphQL information in that json payload is different between the two versions. 

Specifically v1 has the `httpMethod` and `path` at the root where v2 has this nested into `requestContext.http`. I'm embarrased to say how long it took me to figure all that out after having changed a value in my SAM template from a copy/pasta. All of the sudden my working event payloads started giving me errors because the apollo-server-lambda actually wasn't getting invoked because the `httpMethod` was bonked.